test/BaselineOutput/Common/TextLoader/missing_fields-with-impute.csv

```diff
@@ -0,0 +1,18 @@
+int,description,num1,num2,date,num3,num4
+0,"this is a description",0.12,0.34,01/01/2001,0.56,0.78
+0,"this has an empty int and date", 1.1, 11.11,1/1/0001,111.111,1111.11111
+0,"this has a quoted empty int and date", 1.1, 11.11,1/1/0001,111.111,1111.11111
+1,"this has a quoted int and date", 1.1, 11.11,1/1/2001,111.111,1111.11111
+2,"this has an empty num1 and a space in num3",NaN,22.22,2/2/2002,NaN,2222.2222
+3,"this has an empty quoted num1 and a quoted space in num3",NaN,33.33,3/3/2003,NaN,3333.3333
+4,"this has a space in num2 and a space in num4",4.4,NaN,4/4/2004,444.444,NaN 
+5,"this has a quoted space num2 and quoted space in num4",5.5,NaN,5/5/2005,555.555,NaN
+// The next two rows map the missing columns as 0, as it was decided not to impute with NaN
+// in this case
+6,"this has no date, num3 or num4 (the separator corresponding to them is also missing)",6.6,66.66,1/1/0001,0,0
+7,"this has no num4 (the separator corresponding to it is missing)",7.7,77.77,7/7/2007,777.777,0
```

Comment:
_(appended to example to allow for threaded discussion)_

The feature I implement here means that "1,,dog" will be loaded as "1,NaN,dog" if the second column is a float/double. And to load "1,dog," as "1,dog,NaN" if the third column is a float/double. This was the behavior requested on #4132 

What to do with incomplete rows was undefined in the request. I.e. what to do if there are supposed to be 6 columns but on one row there's only 2? (I.e. `1,dog`, without the last separator)

In such a case, I took the decision to simply load the missing columns with default values (i.e. 0 for Float and Double) even when using the new feature of this PR.

The reason behind this is that the implementation required to load those missing columns as NaNs only for float/doubles would be somewhat hacky (and I didn't figure out how to make it work for vector types which included fields on missing columns on a given row). Besides, since it might be a fringe case. it's perhaps better not to support it.

Any thoughts on this, @justinormont @harishsk ?